deepchem/models/torch_models/weavemodel_pytorch.py

```diff
@@ -0,0 +1,593 @@
+            with the i-th atom belongs to.
+        atom_to_pair: np.ndarray
+            Of shape `(N_pairs, 2)`. The i-th row in this array will be the array
+            `[a, b]` if `(a, b)` is a pair to be considered. (Note by symmetry, this
+            implies some other row will contain `[b, a]`.
+        """
+        atom_feat = []
+        pair_feat = []
+        atom_split = []
+        atom_to_pair = []
+        pair_split = []
+        start = 0
+        for im, mol in enumerate(X_b):
+            n_atoms = mol.get_num_atoms()
+            # pair_edges is of shape (2, N)
+            pair_edges = mol.get_pair_edges()
+            # number of atoms in each molecule
+            atom_split.extend([im] * n_atoms)
+            # index of pair features
+            C0, C1 = np.meshgrid(np.arange(n_atoms), np.arange(n_atoms))
+            atom_to_pair.append(pair_edges.T + start)
+            # Get starting pair atoms
+            pair_starts = pair_edges.T[:, 0]
+            # number of pairs for each atom
+            pair_split.extend(pair_starts + start)
+            start = start + n_atoms
+
+            # atom features
+            atom_feat.append(mol.get_atom_features())
+            # pair features
+            pair_feat.append(mol.get_pair_features())
+
+        return (np.concatenate(atom_feat, axis=0),
+                np.concatenate(pair_feat, axis=0), np.array(pair_split),
+                np.array(atom_split), np.concatenate(atom_to_pair, axis=0))
+
+    def default_generator(
+            self,
+            dataset: Dataset,
+            epochs: int = 1,
+            mode: str = 'fit',
+            deterministic: bool = True,
+            pad_batches: bool = True) -> Iterable[Tuple[List, List, List]]:
+        """Convert a dataset into the tensors needed for learning.
+
+        Parameters
+        ----------
+        dataset: `dc.data.Dataset`
+            Dataset to convert
+        epochs: int, optional (Default 1)
+            Number of times to walk over `dataset`
+        mode: str, optional (Default 'fit')
+            Ignored in this implementation.
+        deterministic: bool, optional (Default True)
+            Whether the dataset should be walked in a deterministic fashion
+        pad_batches: bool, optional (Default True)
+            If true, each returned batch will have size `self.batch_size`.
+
+        Returns
+        -------
+        Iterator which walks over the batches
+        """
+        for epoch in range(epochs):
+            for (X_b, y_b, w_b,
+                 ids_b) in dataset.iterbatches(batch_size=self.batch_size,
+                                               deterministic=deterministic,
+                                               pad_batches=pad_batches):
+                if y_b is not None:
+                    if self.model.mode == 'classification':
+                        y_b = to_one_hot(y_b.flatten(),
```

Comment:
separate handling of labels in the data with in the default generator may not be required

Reply:
please check

Reply:
default_generator function gets called internally for Numpy Dataset when model.fit() is run. And to keep it close to tensorflow implementation I thought it would be best to keep the default generator function.